src/Registration/User.ts

```diff
@@ -0,0 +1,11 @@
+export type UserFirstName = string;
+export type UserSurname = string;
+export type UserEmail = string;
+export type UserType = string;
```

Comment:
You don't need to do all these types - It's very verbose, and doesn't add anything.
Just change the below `User` type to have them as strings...

```ts
export type User = {
  firstName: string;
  surname: string;
  email: string;
  type: UserType; (the enum)
}
```